packages/infra/lib/api-stack/terminology-server-service.ts

```diff
@@ -0,0 +1,162 @@
+import { Duration, StackProps } from "aws-cdk-lib";
+import * as cloudwatch from "aws-cdk-lib/aws-cloudwatch";
+import { SnsAction } from "aws-cdk-lib/aws-cloudwatch-actions";
+import * as ec2 from "aws-cdk-lib/aws-ec2";
+import * as ecr_assets from "aws-cdk-lib/aws-ecr-assets";
+import * as ecs from "aws-cdk-lib/aws-ecs";
+import { FargateService } from "aws-cdk-lib/aws-ecs";
+import * as ecs_patterns from "aws-cdk-lib/aws-ecs-patterns";
+import * as s3 from "aws-cdk-lib/aws-s3";
+import { Bucket } from "aws-cdk-lib/aws-s3";
+import { Construct } from "constructs";
+import { EnvConfig } from "../../config/env-config";
+import { getConfig } from "../shared/config";
+import { vCPU } from "../shared/fargate";
+import { MAXIMUM_LAMBDA_TIMEOUT } from "../shared/lambda";
+import { buildLbAccessLogPrefix } from "../shared/s3";
+import { addDefaultMetricsToTargetGroup } from "../shared/target-group";
+import { isProd } from "../shared/util";
+
+export function settings() {
+  const config = getConfig();
+  const prod = isProd(config);
+  // Watch out for the combination of vCPUs and memory, more vCPU requires more memory
+  // https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#task_size
+  const cpuAmount = prod ? 4 : 2;
+  return {
+    cpuAmount,
+    cpu: cpuAmount * vCPU,
+    memoryLimitMiB: prod ? 8192 : 4096,
+    taskCountMin: prod ? 1 : 1,
+    taskCountMax: prod ? 10 : 2,
+    // How long this service can run for
+    maxExecutionTimeout: MAXIMUM_LAMBDA_TIMEOUT,
+  };
+}
+
+interface TermServerServiceProps extends StackProps {
+  config: EnvConfig;
+  version: string | undefined;
+  generalBucket: Bucket;
+}
+
+export function createTermServerService(
+  stack: Construct,
+  props: TermServerServiceProps,
+  vpc: ec2.IVpc,
+  alarmAction: SnsAction | undefined
+): { service: FargateService; address: string } {
+  const { cpu, memoryLimitMiB, taskCountMin, taskCountMax, maxExecutionTimeout } = settings();
+
+  const bucketName = "umls-terminology-db";
+  // const terminologyBucket = new s3.Bucket(stack, "TerminologyBucket", {
+  //   bucketName,
+  //   publicReadAccess: false,
+  //   encryption: s3.BucketEncryption.S3_MANAGED,
+  //   versioned: true,
+  // });
+
+  const terminologyBucket = s3.Bucket.fromBucketName(stack, "TerminologyBucket", bucketName);
```

Comment:
before PR is approved, I can change this back to be a create Bucket instead of a from Bucket. Used from since it was easier for dev and I didnt have to deal with permissions to delete the DB before each branch to staging to stop resource conflicts

Reply:
Do we expect to store lots of files for the term server? From what I can tell, no, right? Then I'd suggest using the `metriport-multi-purpose-<env>` one.